### PR TITLE
Turn exercise 003 into practice

### LINKED ISSUE #2167: Turn exercise 003 into practice
Extract the OCaml code from the markdown file corresponding to this exercise. The file is: `data/exercise/003_NAME.md` (where `003` is the number from the title of this issue and `NAME` is the rest of the file name, for instance `data/exercise/001_tail.md` is the file which corresponds to exercise `001`). Use the code in this file to create a folder `practice/003` (using this same number).

Populate the `003` folder with (almost) the same files as in folders `practice/001` and `practice/002` (they are examples).

Three files are different in each folder:
1. `practice/003/ex.ml`: it contains the function's signature, which needs to be implemented by the learner; 
1. `practice/003/work/impl.ml`: it includes the template meant to be filed by the learner;
1. `practice/003/answer/impl.ml`: consists of the solution from the original markdown file.

The examples included in the original markdown file must be turned into test cases in the `practice/003/ex.ml` file.

Folder structure to create:
```
practice/003/
├── answer
│   ├── impl.ml
│   └── test
│       ├── dune
│       └── run.ml
├── dune
├── dune-project
├── ex.ml
└── work
    ├── impl.ml
    └── test
        ├── dune
        └── run.ml
```

### PR DESCRIPTION
**Pull request by mnaibei**
This PR is in reference to task #2167

I am still confused if this is the proper implementation based on task description, I would appreciate detailed feedback on my attempt if possible @SaySayo  @cuihtlauac 

PS I wanted this to be a draft PR but I forgot to change that, please consider it DRAFT

Fix Issue #2167

### CODE REVIEW

**Comment by cuihtlauac** [comment]
Thanks @mnaibei, you are doing well
- You are missing file `dune` in folders `work/test` and `answer/test`
- Several files are missing newline at end of file
- The following commands must execute successfully:
  - `dune built --root .`
  - `dune test --root answer`

File: .gitignore
Comment (cuihtlauac) [inline .gitignore:1]: This file should not be part of the PR

File: practice/003/answer/impl.ml
Comment (cuihtlauac) [inline practice/003/answer/impl.ml:1]: Add newline at eof

File: practice/003/work/impl.ml
Comment (cuihtlauac) [inline practice/003/work/impl.ml:1]: Add newline at eof

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

File: practice/003/work/impl.ml
Comment (mnaibei) [reply practice/003/work/impl.ml:1]: @cuihtlauac yes I will do this asap., I am currently facing internet issues

**Comment by mnaibei** [comment]
Hi @cuihtlauac in my local file I have put in the required EOF newline but once pushed the newline is removed🤔., could you advise me on how to prevent this?

here is a screenshot of one of the files:
Screenshot 2024-03-14 at 18 51 00

File: .gitignore
Comment (cuihtlauac) [inline .gitignore:1]: You should not modify this file in this PR

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Pull request closed without merging**